### PR TITLE
Resolve Issue #17

### PR DESCRIPTION
**Pull request by LeMoussel**
Use `retry` crate to try a few times with a delay between each attempt in `export` function.

### CODE REVIEW

**Comment by nastevens** [comment]
It looks like you've got some tabs mixed in with spaces and that's messing up the formatting on GitHub. Could you please replace all tabs with 4 spaces?

File: src/export.rs [hunk 1]
```diff
@@ -73,7 +73,11 @@ pub fn export(pin_config: &PinConfig, symlink_root: Option<&str>) -> Result<(),
         try!(fs::create_dir_all(symroot));
 
         // set the pin direction
-        try!(pin_config.get_pin().set_direction(pin_config.direction.clone()));
+		    // Call up to 3 times, waiting 100 milliseconds after each unsuccessful attempt.
+        match retry(3, 100, || pin_config.get_pin().set_direction(pin_config.direction.clone()), |res| res.is_ok()) {
```
Comment (nastevens) [inline src/export.rs]: A couple things on this line:

- Please pull out the constants with names like `PIN_CONFIGURE_RETRIES` and `PIN_CONFIGURE_TIMEOUT`
- This should probably be split across multiple lines, it's getting a bit long. [`rustfmt`](https://crates.io/crates/rustfmt) will do this automatically for you.

**Comment by nastevens** [comment]
The build is also failing - you need to add `retry` to `lib.rs` as an `extern crate`, and also `use retry::retry` in `export.rs`

**Comment by Disasm** [comment]
Ping from triage: @LeMoussel, could you rebase and fix your changes according to the comments?

**Comment by mathk** [comment]
@rust-embedded/embedded-linux Will someone take that work or should we close as no response from the author?

**Comment by nastevens** [comment]
@mathk This can just be closed. It's not clear to me on second review that retrying this one operation is the right thing to do anyway - an error would return an error code from the tool and the operation can be retried from the caller.

**Pull request closed without merging**